src/FocusApp.Client/Methods/User/EditUserSelectedIsland.cs

```diff
@@ -0,0 +1,76 @@
+﻿using FocusApp.Client.Clients;
+using FocusApp.Client.Helpers;
+using FocusApp.Shared.Data;
+using FocusApp.Shared.Models;
+using FocusCore.Commands.User;
+using FocusCore.Responses;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+
+namespace FocusApp.Methods.User;
+public class EditUserSelectedIsland
+{
+    public class Response
+    {
+        public string Message { get; set; }
+        public bool IsSuccessful { get; set; }
+    }
+
+    internal class Handler : IRequestHandler<EditUserSelectedIslandCommand, MediatrResult>
+    {
+        FocusAppContext _localContext;
+        ILogger<Handler> _logger;
+        IAuthenticationService _authenticationService;
+        IAPIClient _client;
+        public Handler(FocusAppContext localContext, ILogger<Handler> logger, IAuthenticationService authenticationService, IAPIClient client)
+        {
+            _localContext = localContext;
+            _logger = logger;
+            _authenticationService = authenticationService;
+            _client = client;
+        }
+
+        public async Task<MediatrResult> Handle(EditUserSelectedIslandCommand command, CancellationToken cancellationToken)
+        {
+            try
+            {
+                // Update the user on the server
+                await _client.EditUserSelectedIsland(command, cancellationToken);
+
+                // Update the local database to reflect the changes made to the user
+                Shared.Models.User? user = await _localContext.Users.FirstOrDefaultAsync(u => u.Id == command.UserId, cancellationToken);
+
+                // Fetch island from local db if not null
+                if (command.IslandId != null)
+                {
+                    Island? island = await _localContext.Islands.Where(i => command.IslandId == i.Id).FirstOrDefaultAsync(cancellationToken);
```

Comment:
No need to delay PR for this, but just a note for future development: the FirstOrDefaultAsync function can take in a lambda expression that can negate the need for the Where clause!

Ex./ _localContext.Islands.FirstOrDefaultAsync(i => command.IslandId == i.Id, cancellationToken) is an equivalent statement to what you have written.

No need to make this adjustment, just wanted to shout it out!